packages/mdc-icon-button/foundation.js

```diff
@@ -0,0 +1,251 @@
+
+    /** @private {boolean} */
+    this.on_ = false;
+
+    /** @private {boolean} */
+    this.disabled_ = false;
+
+    /** @private {number} */
+    this.savedTabIndex_ = -1;
+
+    /** @private {?IconToggleState} */
+    this.toggleOnData_ = null;
+
+    /** @private {?IconToggleState} */
+    this.toggleOffData_ = null;
+
+    this.clickHandler_ = /** @private {!EventListener} */ (
+      () => this.toggleFromEvt_());
+
+    /** @private {boolean} */
+    this.isHandlingKeydown_ = false;
+
+    this.keydownHandler_ = /** @private {!EventListener} */ ((/** @type {!KeyboardKey} */ evt) => {
+      if (isSpace(evt)) {
+        this.isHandlingKeydown_ = true;
+        return evt.preventDefault();
+      }
+    });
+
+    this.keyupHandler_ = /** @private {!EventListener} */ ((/** @type {!KeyboardKey} */ evt) => {
+      if (isSpace(evt)) {
+        this.isHandlingKeydown_ = false;
+        this.toggleFromEvt_();
+      }
+    });
+  }
+
+  init() {
+    this.refreshToggleData();
+    this.savedTabIndex_ = this.adapter_.getTabIndex();
+    this.adapter_.registerInteractionHandler('click', this.clickHandler_);
+    this.adapter_.registerInteractionHandler('keydown', this.keydownHandler_);
+    this.adapter_.registerInteractionHandler('keyup', this.keyupHandler_);
+  }
+
+  refreshToggleData() {
+    const {DATA_TOGGLE_ON, DATA_TOGGLE_OFF} = MDCIconButtonToggleFoundation.strings;
+    this.toggleOnData_ = this.parseJsonDataAttr_(DATA_TOGGLE_ON);
+    this.toggleOffData_ = this.parseJsonDataAttr_(DATA_TOGGLE_OFF);
+  }
+
+  destroy() {
+    this.adapter_.deregisterInteractionHandler('click', this.clickHandler_);
+    this.adapter_.deregisterInteractionHandler('keydown', this.keydownHandler_);
+    this.adapter_.deregisterInteractionHandler('keyup', this.keyupHandler_);
+  }
+
+  /** @private */
+  toggleFromEvt_() {
+    this.toggle();
+    const {on_: isOn} = this;
+    this.adapter_.notifyChange(/** @type {!IconButtonToggleEvent} */ ({isOn}));
+  }
+
+  /** @return {boolean} */
+  isOn() {
+    return this.on_;
+  }
+
+  /** @param {boolean=} isOn */
+  toggle(isOn = !this.on_) {
+    this.on_ = isOn;
+
+    const {ARIA_LABEL, ARIA_PRESSED} = MDCIconButtonToggleFoundation.strings;
+
+    if (this.on_) {
+      this.adapter_.setAttr(ARIA_PRESSED, 'true');
+    } else {
+      this.adapter_.setAttr(ARIA_PRESSED, 'false');
+    }
+
+    const {cssClass: classToRemove} =
+        this.on_ ? this.toggleOffData_ : this.toggleOnData_;
+
+    if (classToRemove) {
+      this.adapter_.removeClass(classToRemove);
+    }
+
+    const {content, label, cssClass} = this.on_ ? this.toggleOnData_ : this.toggleOffData_;
+
+    if (cssClass) {
+      this.adapter_.addClass(cssClass);
+    }
+    if (content) {
+      this.adapter_.setText(content);
+    }
+    if (label) {
+      this.adapter_.setAttr(ARIA_LABEL, label);
+    }
+  }
+
+  /**
+   * @param {string} dataAttr
+   * @return {!IconToggleState}
+   */
+  parseJsonDataAttr_(dataAttr) {
```

Comment:
Log the data attribute redesign as a follow-up issue... but we should really do that before merging this to master and publishing this package to avoid immediate breaking changes.

It should be possible for you to create a centralized branch for mdc-icon-button changes and edit this PR to target that branch.